spec/mailers/user_mailer_spec.rb

```diff
@@ -60,4 +60,122 @@
       expect(mail.body).to include(reset_password_internal_user_url(user, token: user.reset_password_token))
     end
   end
+
+  describe '#got_new_comment' do
+    let(:user) { create(:learner) }
+    let(:token) { AuthenticationToken.find_by(user: user) }
+    let(:request_for_comment) { create(:rfc_with_comment, user: user) }
+    let(:commenting_user) { InternalUser.create(attributes_for(:teacher)) }
+    let(:mail) { described_class.got_new_comment(request_for_comment.comments.first, request_for_comment, commenting_user).deliver_now }
+
+    it 'sets the correct sender' do
+      expect(mail.from).to include(CodeOcean::Application.config.action_mailer[:default_options][:from])
+    end
+
+    it 'sets the correct subject' do
+      expect(mail.subject).to eq(I18n.t('mailers.user_mailer.got_new_comment.subject', commenting_user_displayname: commenting_user.displayname))
+    end
+
+    it 'sets the correct receiver' do
+      expect(mail.to).to include(request_for_comment.user.email)
+    end
+
+    it 'includes the correct URL' do
+      expect(mail.body).to include(request_for_comment_url(request_for_comment))
```

Comment:
What URL is expected here? Same argument as above. The authentication token is supposed to be part of this URL, isn't it? Currently, it could appear anywhere in the e-mail body. A regular expression can help here.